### PR TITLE
Bug 1172349 - [Settings] Convert root panel switches to use gaia-switch

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1172349

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0690d3466682ff68535b3596ed5c3d45b9f894aa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a180eedf763f0b517b82ceb72bc0abf557c5e45d

File: apps/settings/style/icons.css [hunk 1]
```diff
@@ -27,6 +27,14 @@ a.menu-item:active:after {
   color: #91aab0;
 }
 
+gaia-switch .menu-item[data-icon] {
+  -moz-padding-start: 4rem;
+}
+
+gaia-switch .menu-item[data-icon] ~ details {
+ -moz-padding-start: 4rem;
```
Comment (gasolin) [inline apps/settings/style/icons.css]: need one more space

File: apps/settings/index.html [hunk 2]
```diff
@@ -15,7 +15,6 @@
 
     <!-- Common style -->
     <link rel="stylesheet" type="text/css" href="shared/style/headers.css">
-    <link rel="stylesheet" type="text/css" href="shared/style/switches.css">
```
Comment (gasolin) [inline apps/settings/index.html]: \o/

File: apps/settings/test/marionette/app/regions/root.js [hunk 3]
```diff
@@ -96,7 +99,7 @@ RootPanel.prototype = {
   // geolocation
   geolocation: function(enabled) {
     if (enabled !== this.geolocationCheckboxChecked) {
-      this.waitForElement('geolocationSpan').tap();
+      this.waitForElement('geolocationCheckbox').tap();
```
Comment (gasolin) [inline apps/settings/test/marionette/app/regions/root.js]: .click()?

Comment (KevinGrandon) [inline apps/settings/test/marionette/app/regions/root.js]: .tap() seemed to be fine here, so maybe it's not needed? I think in most cases it probably doesn't matter...

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=52d2a9105752f3b863164275980402b2c0688f9d

File: apps/settings/test/marionette/app/regions/root.js (hunk 3 above)
Comment (gasolin) [inline apps/settings/test/marionette/app/regions/root.js]: its fine if test passed :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=31b6fc9e5a6b8f9b7ec5a25de4a41f78d4a265ef

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=574132a4f159412e98566abd552da93b00f3ae1b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4dfd74b4a2f5e083ecd97295e72ef9986dd42f4d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6e4607a63216280d39c7516c65409580d4f57ad6

**Pull request merged**